Mathlib/Topology/Algebra/Module/FiniteDimension.lean

```diff
@@ -54,9 +54,10 @@ noncomputable section
 
 section Field
 
-variable {𝕜 E F : Type*} [Field 𝕜] [TopologicalSpace 𝕜] [AddCommGroup E] [Module 𝕜 E]
-  [TopologicalSpace E] [AddCommGroup F] [Module 𝕜 F] [TopologicalSpace F] [IsTopologicalAddGroup F]
-  [ContinuousSMul 𝕜 F]
+variable {𝕜 E F : Type*} [Field 𝕜]
```

Comment:
This generalizes to `CommRing` as
```lean-4
/-- The space of continuous linear maps between finite-dimensional spaces is finite-dimensional. -/
instance [Module.Finite 𝕜 E] [Module.Finite 𝕜 F] [IsNoetherian 𝕜 F] : Module.Finite 𝕜 (E →L[𝕜] F) :=
  .of_injective (ContinuousLinearMap.coeLM 𝕜 : (E →L[𝕜] F) →ₗ[𝕜] E →ₗ[𝕜] F)
    ContinuousLinearMap.coe_injective
```
I guess we should keep around the existing instance (proven with `inferInstance` or the `Module.Finite` one directly) for the sake of matching `FiniteDimensional` with loogle.

Reply:
If we're worried about performance, let's add my suggestion as a `theorem` so it can be found by somehow who needs it with Loogle?